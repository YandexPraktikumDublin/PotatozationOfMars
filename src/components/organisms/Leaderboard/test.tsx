import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { Store } from 'redux'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

import { Leaderboard } from '.'

const mockStore = configureStore([])

describe('<Leaderboard />', () => {
  let store: Store

  const initialState = {
    leaderboard: []
  }

  beforeEach(() => {
    store = mockStore(initialState)
  })

  it('should renders correct <Leaderboard />', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Leaderboard />
      </Provider>
    )

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
