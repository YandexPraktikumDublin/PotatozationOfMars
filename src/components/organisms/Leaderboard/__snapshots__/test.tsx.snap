// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Leaderboard /> should renders correct <Leaderboard /> 1`] = `
<div
  className="text-primary dark:text-white"
>
  <List>
    <NameValueListItem
      key="IvanIvanov"
      name="1 IvanIvanov"
      value={1023}
    />
    <NameValueListItem
      key="PetrPetrov"
      name="2 PetrPetrov"
      value={1010}
    />
  </List>
</div>
`;
